code/modules/admin/verbs/playsound.dm

```diff
@@ -127,3 +129,106 @@ GLOBAL_LIST_EMPTY(sounds_cache)
 		SEND_SOUND(M, sound(null))
 		var/client/C = M.client
 		C?.tgui_panel?.stop_music()
+
+/client/proc/play_web_sound()
+	set category = "Event"
+	set name = "Play Internet Sound"
+	if(!check_rights(R_SOUNDS))
+		return
+
+	if(!GLOB.configuration.system.ytdlp_url)
+		to_chat(src, "<span class='boldwarning'>yt-dlp was not configured, action unavailable</span>") //Check config
+		return
+
+
+	var/web_sound_input = tgui_input_text(src, "Enter content URL (supported sites only, leave blank to stop playing)", "Play Internet Sound", null)
+	if(istext(web_sound_input))
```

Comment:
I don't trust reading indents on a phone but if the entire proc is indented after this if, surely its better to guard clause it?

Reply:
It's not completely indented after that if statement. There is an else statement that returns a var to stop sounds if there is no input text and another if statement to check for HTTPS in the URL (if there is no https it returns) and finally a last if statement to actually send data to tgchat or stop sounds (this should catch everything)

yeah it's not the prettiest code in the world

Reply:
nevermind I am just dumb and was looking at the wrong if statement.

I added a return to the end of the proc on that if statement